File: member_crud-jpa/app/src/main/java/sample/member/app/security/SimpleUser.java
package sample.member.app.security;

import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

@Getter
public class SimpleUser implements UserDetails {
	private String username;
	private Collection<? extends GrantedAuthority> authorities;
	private String password;
	private boolean accountNonExpired;
	private boolean accountNonLocked;
	private boolean credentialsNonExpired;
	private boolean enabled;

	public SimpleUser(String username, Collection<? extends GrantedAuthority> authorities, String password) {
		this.username = username;
		this.authorities = authorities;
		this.password = password;
		this.accountNonExpired = true;
		this.accountNonLocked = true;
		this.credentialsNonExpired = true;
		this.enabled = true;
	}
}

File: 2017-multi-jpa-kotlin/app/src/main/kotlin/sample/member/app/security/SimpleUser.kt
package sample.member.app.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

//좀 더러운 해결법
//https://youtrack.jetbrains.com/issue/KT-6653
class SimpleUser(private val _username: String, private val _authorities: Collection<GrantedAuthority>, private val _password: String) : UserDetails {
	override fun getAuthorities(): Collection<GrantedAuthority> = _authorities
	override fun getUsername(): String = _username
	override fun getPassword(): String = _password

	override fun isAccountNonExpired(): Boolean = true
	override fun isEnabled(): Boolean = true
	override fun isCredentialsNonExpired(): Boolean = true
	override fun isAccountNonLocked(): Boolean = true
}
